src/v/bytes/bytes.h

```diff
@@ -16,22 +16,169 @@
 
 #include <seastar/core/sstring.hh>
 
-#include <absl/hash/hash.h>
+#include <absl/container/inlined_vector.h>
 
+#include <algorithm>
 #include <cstdint>
 #include <iosfwd>
 #include <span>
 
-// cannot be a `std::byte` because that's not sizeof(char)
+class bytes_view;
+
 constexpr size_t bytes_inline_size = 31;
-using bytes = ss::basic_sstring<
-  uint8_t,  // Must be different from char to not leak to std::string_view
-  uint32_t, // size type - 4 bytes - 4GB max - don't use a size_t or any 64-bit
-  bytes_inline_size, // short string optimization size
-  false              // not null terminated
-  >;
-
-using bytes_view = std::basic_string_view<uint8_t>;
+
+class bytes {
```

Comment:
Is there a size difference in the struct between the two? I would hope not, but we should check.

Reply:
Looks like boost::small_vector is the same size as absl too.

Reply:
<img width="664" alt="Screenshot 2024-09-20 at 1 23 00 PM" src="https://github.com/user-attachments/assets/312e63b0-ab64-4c04-a531-8505165be49c">

Ugh github dropped this image.

It looks like absl::InlineVector is 40 bytes and ss::sstring is 32 bytes. Probably because seastar is taking advantage of the high bits in a pointer or something.